sw/device/lib/base/mmio.h

```diff
@@ -234,6 +234,62 @@ inline void mmio_region_nonatomic_set_mask32(mmio_region_t base,
   mmio_region_write32(base, offset, value);
 }
 
+/**
+ * Masked Register field.
+ */
+typedef struct mmio_region_field32 {
+  uint32_t mask;  /**< The field mask. */
+  uint32_t index; /**< The field position within the a register. */
+  uint32_t value; /**< The field value to be written. */
+} mmio_region_field32_t;
+
+/**
+ * Sets the |field| from the MMIO region |base| at the given offset.
+ *
+ * This function performs a non-atomic read-write-modify operation on a
+ * MMIO region.
+ *
+ * @param base the region to set the field in.
+ * @param offset the offset to set the field at, in bytes.
+ * @param field field within selected register field to be set.
+ */
+inline void mmio_region_nonatomic_set_field32(mmio_region_t base,
+                                              ptrdiff_t offset,
+                                              mmio_region_field32_t field) {
+  uint32_t register_value = mmio_region_read32(base, offset);
+  register_value &= ~(field.mask << field.index);
+  register_value |= (field.value & field.mask) << field.index;
```

Comment:
I almost feel like this snippet should be factored out into another inline function...

Reply:
Done in a separate PR as discussed previously. I will link that PR against this one.